test/Table_Tests/src/Expression_Spec.enso

```diff
@@ -0,0 +1,156 @@
+
+    Test.group "Expression Text literals" <|
+        specify_test "should be able to add a text column" expression_test->
+            expression_test "'Hello World'" 'Hello World'
+            expression_test "'Hello \'World\''" "Hello 'World'"
+            expression_test '"Hello World"' 'Hello World'
+            expression_test '"Hello ""World"""' 'Hello "World"'
+
+    Test.group "Expression Text literals" <|
+        specify_test "should be able to get a Column" expression_test->
+            expression_test "[A]" (column_a.at 1)
+            expression_test "[Bad]] Name]" (column_odd.at 1)
+
+    Test.group "Expression Nothing literals" <|
+        specify_test "should be able to add an nothing column" expression_test->
+            expression_test "null" Nothing
+            expression_test "Nothing" Nothing
+
+    Test.group "Expression Date and Time literals" <|
+        specify_test "should be able to add a date or time column" expression_test->
+            expression_test "#2020-12-23#" (Date.new 2020 12 23)
+            expression_test "#12:34#" (Time_Of_Day.new 12 34)
+            expression_test "#12:34:56#" (Time_Of_Day.new 12 34 56)
+            expression_test "#12:34:56.789#" (Time_Of_Day.new 12 34 56 789000000)
+            expression_test "#12:34:56.789000123#" (Time_Of_Day.new 12 34 56 789000123)
+            expression_test "#2020-12-23 12:34#" (Date_Time.new 2020 12 23 12 34)
+            expression_test "#2020-12-23 12:34:56#" (Date_Time.new 2020 12 23 12 34 56)
+            expression_test "#2020-12-23 12:34:56+02:30#" (Date_Time.new 2020 12 23 10 04 56)
+            expression_test "#2020-12-23 12:34:56Z[UTC]#" (Date_Time.new 2020 12 23 12 34 56 zone=Time_Zone.utc)
+            expression_test "#2020-12-23 12:34:56.157+01[UTC]#" (Date_Time.new 2020 12 23 11 34 56 157000000 zone=Time_Zone.utc)
+            expression_test "#2020-12-23T12:34[Europe/Warsaw]#" (Date_Time.new 2020 12 23 12 34 zone=Time_Zone.parse("Europe/Warsaw"))
+
+    Test.group "Expression Arithmetic" <|
+        specify_test "should be able to do basic arithmetic" expression_test->
+            expression_test "1+1" 2
+            expression_test "23-15" 8
+            expression_test "2.5*4.2" 10.5
+            expression_test "1_000.456/2" 500.228
+            expression_test "2^4" 16
+            expression_test "11%3" 2
+
+        specify_test "should be able to do basic arithmetic with order" expression_test->
+            expression_test "1+1*2+2" 5
+            expression_test "23-15/3+6" 24
+            expression_test "52.92/4.2^2" 3
+            expression_test "(1+1)*2+2" 6
+
+        specify_test "should be able to do basic arithmetic with whitespace" expression_test->
+            expression_test "1 + 1" 2
+            expression_test "  23 -15  " 8
+            expression_test "2.5*  4.2" 10.5
+            expression_test "1_000.456/   2" 500.228
+            expression_test " 2 ^ 4 " 16
+            expression_test "  11   % 3    " 2
```

Comment:
I'd add a test that checks if here whitespace _does not_ influence order of operations (contrary to regular Enso code).